frontend/src/profile/progress/ProgressTab.tsx

```diff
@@ -79,6 +83,46 @@ const ProgressTab: React.FC<ProgressTabProps> = ({ user, isCurrentUser }) => {
             }
         });
 
+        const suggestedTasks: Category = {
+            name: RequirementCategory.SuggestedTasks,
+            requirements: [],
+            totalComplete: 0,
+            totalRequirements: 0,
+        };
+
+        const desiredTaskCount = 3;
+
+        const recentRequirements = Object.values(user.progress)
+            .toSorted((a, b) => b.updatedAt.localeCompare(a.updatedAt))
+            .map((progress) => requirementsById[progress.requirementId])
+            .filter((r) => !!r && !isComplete(cohort, r, user.progress[r.id]));
+
+        suggestedTasks.requirements = recentRequirements.slice(0, desiredTaskCount - 1);
+
+        const otherIncompleteRequirements = requirements.filter(
+            (r) =>
+                !isComplete(cohort, r, user.progress[r.id]) &&
+                suggestedTasks.requirements.findIndex((recent) => recent.id === r.id) < 0,
+        );
+
+        const now = new Date();
+        const daysSinceEpoch = Math.floor(now.getTime() / 8.64e7);
+        const idx = daysSinceEpoch % requirements.length;
+
+        // Fill the remaining suggest tasks slots with tasks that rotate day by day.
+        suggestedTasks.requirements = [
+            ...suggestedTasks.requirements,
+            ...[
+                ...otherIncompleteRequirements.slice(idx),
+                ...otherIncompleteRequirements.slice(0, idx),
```

Comment:
I'm not sure I like this slicing approach. This will basically create a sliding window over the requirements that cycles one by one through all the incomplete/non-recent requirements each day. So someone who logs in 3 days in a row would for example get 3 Games + Analysis tasks in a row

Reply:
For example, suppose the first time I log in, idx happens to point to the first games and analysis tasks. I haven't done any tasks, so I will see Play 60 Classical Games, Annotate 60 Classical Games and Complete 20 Postmortems as my 3 suggested tasks.

Reply:
I think we should instead use a seedable random number generator (here is one package, although there may be better ones https://www.npmjs.com/package/random-seedable) and seed it with the daysSinceEpoch value. Then use that to choose the proper number of random tasks. This will keep the suggested random tasks stable across re-loads and re-renders over the course of a single day, but should improve the randomness